src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] elements;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    private void resize() {
+        capacity = (int) (capacity * 1.5);
+        elements = Arrays.copyOf(elements, capacity);
```

Comment:
Using `Arrays.copyOf` violates the checklist requirement: 'Don't use Objects, Arrays, or any other util class.' Consider implementing your own array copying logic using `System.arraycopy()` instead .